broadinstitute/pearl
# JN-456: Enabling live-editing of site content

Enables quick-and-dirty demo of live editing of SiteContent.  

https://localhost:3000/ourhealth/env/sandbox/siteContent

Saving is currently not supported, nor is anything beyond simple text edits within the json blobs.  this is only intended for demo purposes.   this moves the landing page components from ui-participant into UI core.  Another thought was having ui-admin depend on ui-participant, but Nick and I discussed and that would probably be unwieldy from a build artifact point of view.   Most of the change is just the dependency wrangling to sort that out.  

Later steps (which might not happen right away depending on demand), include:
1. pulling config fields into separate editable components
2. enabling actual saving
3. add/remove/reorder sections
4. add/remove/reorder navbar items
5. configuring footer, navlogo, and other properties of sitecontent


TO TEST:
1. you'll probably need to stop all ui projects, run `npm install`, and then restart them all (first building ui-core)
2. click through https://sandbox.ourhealth.localhost:3001 to confirm stuff still generally works
3. go to https://localhost:3000/ourhealth/env/sandbox/siteContent.  
4. edit some text, confirm the updates show in the preview panels
